src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -154,10 +154,19 @@ void BaseCouplingScheme::initialize(double startTime, int startTimeWindow)
 
   initializeImplementation();
 
-  if (sendsInitializedData()) {
+  if (_sendsInitializedData) {
     requireAction(constants::actionWriteInitialData());
   }
 
+  if (not _sendsInitializedData && not _receivesInitializedData) {
+    if (isImplicitCouplingScheme()) {
+      if (not doesFirstStep()) {
+        storeDataInWaveforms();
+        moveToNextWindow();
+      }
+    }
+  }
+
```

Comment:
I think this is an interesting change: Since the state is now tracked internally in the Waveform, we also have to tell the Waveform "initialization is done and next thing is advance" before we enter advance the first time. 

* If we call `initializeData()` this is already done, since otherwise the initial data will be lost (see https://github.com/precice/precice/pull/1098/files#diff-ad728b4925806f3679256c80a1006598dd5e4f4ec5c22fe90121729abad234c2R175-R181).
* If we don't call `initializeData()`, this part is missing and, therefore, we have to take care of this in `initialize()` (the last thing happening before calling `advance()`.

It might be a good idea to factor this out into a new function `preAdvance()` that may only be called once and which must be called directly before `advance()`. Any opinion on this @uekerman?

Reply:
`preAdvance()` sounds complicated :see_no_evil: 
In principle, we know what happens during configuration already. You could there pass this information to `time` already.

Reply:
Problem I see during configuration: If there is data, we don't know it yet. Additionally, we also do not know the mesh / number of unknowns yet.